asyncy/omg/ServiceContract.py

```
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+from .OmgExceptions import OmgMismatchedPropertyLengthError, \
+    OmgMissingKeyInBodyError, OmgPropertyKeyMissingTypeError, \
+    OmgPropertyTypeNotFoundError, OmgPropertyValueMismatchError
+
+
+class ServiceContract:
+
+    @classmethod
+    def ensure_value_of_type(cls, _type, value):
+        """
+        Ensures that the value matches the expected type as listed on
+        https://microservice.guide/schema/actions/#arguments.
+
+        Supported types: int, float, number, string, list, map, boolean, or any
+        """
+        if _type == 'string':
+            cls.ensure_type(str, value, _type)
+        elif _type == 'int':
+            cls.ensure_type(int, value, _type)
+        elif _type == 'boolean':
+            cls.ensure_type(bool, value, _type)
+        elif _type == 'map':
+            cls.ensure_type(dict, value, _type)
+        elif _type == 'float':
+            cls.ensure_type(float, value, _type)
+        elif _type == 'list':
+            cls.ensure_type(list, value, _type)
+        elif _type == 'number':
+            cls.ensure_type_list([int, float], value, _type)
+        elif _type == 'any':
+            return
+        else:
+            raise OmgPropertyTypeNotFoundError(_type)
```

This isn't a user exception. This is a platform failure. Hence, this should be an internal assert as we know what types are supported by OMG.

---

i am doing a string compare... so I cannot replace if with assert. But i can do assert type in ['int','float','boolean','string','map','list'] and then let it throw AssertionError?

---

Of course you can:

```py
assert type_ == 'any'
```